services/proxy/pkg/middleware/metrics.go

```diff
@@ -0,0 +1,29 @@
+package middleware
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/go-chi/chi/v5/middleware"
+	"github.com/owncloud/ocis/v2/services/proxy/pkg/metrics"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// Instrumenter provides a middleware to create metrics
+func Instrumenter(m metrics.Metrics) func(next http.Handler) http.Handler {
```

Comment:
Shouldn't we try to use the existing promhttp wrapper to instrumentalize the handler? see https://pkg.go.dev/github.com/prometheus/client_golang/prometheus/promhttp#example-InstrumentRoundTripperDuration

Reply:
discussed. Keeping the small middleware approach with the chi response wrapper.